crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -161,21 +165,15 @@ impl TaskPool {
 
         f(scope_ref);
 
-        // Loop until all tasks are done
-        while executor.try_tick() {}
+        // Wait until the socpe is complete
+        while *pending_tasks.borrow() != 0 {
+            block_on(executor.tick());
+        }
```

Comment:
```suggestion
        block_on(executor.run(async  [
            while *pending_tasks.borrow() != 0 {
                future::yield_now().await;
            }
        }));
```
The current tick loop locks and unlocks a `Mutex` on each tick. The above code should avoid it. Is this feasible as an alternative?

Reply:
This seems to work.